src/App.js

```diff
@@ -116,17 +118,21 @@ export default class App extends Component {
           )}
 
         {this.state.cartridge != null && (
-          <CommonCartridge
-            onHistoryChange={this.handleHistoryChange}
-            src={this.state.cartridge}
-          />
+          <Router>
+            <CommonCartridge
+              onHistoryChange={this.handleHistoryChange}
+              src={this.state.cartridge}
```

Comment:
To keep things dry, can we just do a single Router and have the src as `this.state.cartridge || this.state.file`?

Reply:
That would assume either condition, but I don't think we can. We'll need to handle the absence of both.